bioio_base/standard_metadata.py

```diff
@@ -103,3 +126,104 @@ def to_dict(self) -> dict:
             self.FIELD_LABELS[field]: getattr(self, field)
             for field in self.FIELD_LABELS
         }
+
+
+# Helper functions for OME metadata extraction
+
+
+def get_metadata_element(ome_metadata: OME, path: str) -> Optional[_Element]:
+    """
+    Gets the first occurrence of the given XML path if one exists.
+    """
+    xml = XML(ome_metadata.to_xml())
+    return xml.find(path, OME_NS)
+
+
+def binning(ome_metadata: OME) -> Optional[str]:
+    """
+    Extracts the binning setting from the OME metadata.
+    Returns
+    -------
+    Optional[str]
+        The binning setting as a string. Returns None if not found.
+    """
+    try:
+        el = get_metadata_element(
+            ome_metadata, "./Image/Pixels/Channel/DetectorSettings"
+        )
```

Comment:
this is not the right way to get this info from a OME object.  See the docs for the OME class.  https://ome-types.readthedocs.io/en/latest/API/ome_types.model/#ome_types._autogenerated.ome_2016_06.Channel

Reply:
Converting a OME object to XML is completely defeating the whole purpose of the OME types...

Reply:
>Converting a OME object to XML is completely defeating the whole purpose of the OME types...

I think the purpose is to standardize the way they are formatted less so the access pattern IMO

>this is not the right way to get this info from a OME object. See the docs for the OME class

I'm unsure I care which way this happens - I think this was carried over from how we were accessing this information from XML, but if it is easier to less failure prone to do some from the OME object that seems good to me.

Reply:
Sorry, yeah what matters most is getting the right values out.  It just seemed strange to convert it back to xml when it was converted into this type for this whole reason, to have accessors into a data model rather than using string xml paths.

Reply:
I don't particularly like the xml parsing either. I think its worth trying to go through ome-models, might be a different ticket though...

Reply:
Ok in [5300d36](https://github.com/bioio-devs/bioio-base/pull/38/commits/5300d36a807a4541067370815c8432bcf8c71d53) Now we pull all the information from the OME rather than converting back. Its much nicer that way.